Feature: add mobile navbar

seba1288 opened the pull request:
-added working buton for theme change

Commit:
feature: added light theme navbar

Commit:
feature: added light theme navbar

Commit:
feature: added navbar component

Commit:
feature: added navbar component

szymonbrud commented on src/assets/styles/theme.scss:
```diff
@@ -9,6 +9,10 @@ $breakpoint-tablet: 700px;
 // scheme for color names
 // color-name_color-intensiveness_transparency
 
+$dark-navbar-mobile:rgba(8,0,16);
```
Czy potrzebujemy tych zminnych, bo z tego co widzę to nigdzie nie zostały użyte?

szymonbrud commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+        <div style={{ padding: '0.5rem' }}>
+          <a id="PL">PL</a>
+        </div>
+        <a> | </a>
```
Jeśli jest to zwykły tekst, który nie prowadzi do żadnej zewnętrznej strony, to użyjmy lepiej np. <span>

seba1288 replied:
tu zapomniałeś dac przykład?

szymonbrud replied:
haha, tak sory. Np. <span> w tej sytuacji byłby okej

kamil-ciesla replied:
chyba cos sie nie wyslalo :P

szymonbrud replied:
dobra kolejna próba `"<span>"`

szymonbrud replied:
Mamy toooo 🎉

szymonbrud commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+    }
+  }
+  return (
+    <nav className="MobileNavbar">
```
Staramy się nie używać nazw klas w taki sposób, bo jest duże ryzyko że między komponentami mogły by się powatrzać, co prowadziło by do tego że nasz css byłby kompletnie pomieszany. 

Tutaj w dokumentacji jest opisane w jaki sposób używać scss.
https://github.com/UEK-i-team/documentation/blob/main/best-practices-in-gatsby.MD#5-stylizowanie

kamil-ciesla replied:
troche nie rozumiem sugestii, moglbys podać przykład prawidlowej nazwy?
nie wiem czy chodzi o fakt ze nazwa jest taka sama jak nazwa komponentu, czy moze o PascalCase

szymonbrud replied:
Źle to napisałem, ale już Seba wie o co chodzi
Chodziło o importowanie styli

szymonbrud commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+    <nav className="MobileNavbar">
+      <div className="LogoContainer">
+        <div className="Logo">
+          <img src={logo_icon_light_theme}></img>
```
Zdjęcia statyczne używamy za pomocą elementu `<StaticImage>` który dostarcza gatsby zamiast `<img>`, szybciej się w tedy będą ładowały 😉 Dodaj też prosze artybut `alt`
Link do dokumentacji jak używać:
https://github.com/UEK-i-team/documentation/blob/main/best-practices-in-gatsby.MD#3-u%C5%BCywanie-grafik-w-projekcie

szymonbrud commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+import logo_text_light_theme from '../../assets/images/logo_text_light_theme.png';
+
+export function MobileNavbar() {
+  const [theme, setTheme] = useState(false);
```
Co myślisz aby zmienić nazwę zmiennej `theme`, bo ona trochę mało mówi co się w niej znajduje? Moja propozycja to np. `isDarkTheme`

szymonbrud commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+export function MobileNavbar() {
+  const [theme, setTheme] = useState(false);
+
+  function switchTheme() {
```
1. Ustawianie na początku funkcji `setTheme(false)` a zaraz poniżej sprawdzanie czy jest `false`, nie jest najlepszym podejściem, wiem że działa, ale może się tak zdażyć że czasem nie zadziała. Dodatkowo wyglada nieczytelenie. Proponuję przenieść do `setTheme(false)` na koniec `else`
2. `if (theme == false)` - zawsze stosujemy 3 znaki porównania w if `===` ale tutaj możemy to jeszcze bardziej uprosić i napisać `if(!theme)`
3. Staramy się nie zmianiać wartości JSX za pomocą` document.getElementsByClassName...` robimy to raczej w ostateczności. Staramy się to robić bezpośrednio w JSX.
np. dynamiczna zmiana koloru ikony:  
` <Brightness6RoundedIcon sx={{ color: theme ? "#FFFFFF" : "#000000"}}/>`
np. dynamiczna zmiana koloru elementu: 
`<div className={theme ? TextLight : TextDark}>TUSK TUSK TUSK</div>`
można też zmieniać bezpośrednio css ale lepiej sobie przygotować 2 osobne klasy

szymonbrud commented on src/components/MobileNavbar/MobileNavbar.scss:
```diff
@@ -0,0 +1,88 @@
+}
+
+.Language #ENG {
+    color: rgba(103, 103, 103, 0.668);
```
Mamy zmienną od tego koloru: `$grey_30`

TIP: Gdy zaznaczymy sobie konkretny element w figmie, to po prawej jest napisana nazwa zmiennej koloru, zmienne w figmie pokrywają się ze zmiennymi w kodzie.

szymonbrud reviewed (changes requested):
Ogólnie Seba dobra robota 👍, przetestowałem i fajnie śmiga, ale mam parę uwag i propozycji zmian.

![image](https://github.com/UEK-i-team/team-homepage/assets/42467247/69296685-4031-4b8c-8b2f-ef78296ed026)
Widok musi dobrze wyglądać na każdej rozdzielczości telefonu, podsyłam screena z przykładowej rozdzielczości gdzie nie wygląda to za dobrze i elementy uciekają w prawo. Elementy muszą się ustawiać automatycznie nie mogą być ustawiane statycznie.

Aby sprawdzić czy na każdej rozdzielczości aplikacja zachowuje się poprawnie, polecam wejść w devtoolsy na wyszukiwarce, wybrać sobie parę modeli telefonów oraz pobawić się rozdzielczością ręcznie. Da się tak wyłapać sporo błędów

TheFenix2000 commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+import logo_text_dark_theme from '../../assets/images/logo_text_dark_theme.png';
+import logo_text_light_theme from '../../assets/images/logo_text_light_theme.png';
+
+export function MobileNavbar() {
```
Powinien byc raczej `Navbar` bo potem przecież też będzie na desktopie tylko większy ale ten sam element

TheFenix2000 commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+import '../MobileNavbar/MobileNavbar.scss';
```
Stosujemy klasy z dokumentacji jako moduły {klasa}, trzeba odpowiednio ich póxniej używać, zobacz do innego PR

TheFenix2000 commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+import logo_icon_dark_theme from '../../assets/images/logo_icon_dark_theme.png';
+import logo_icon_light_theme from '../../assets/images/logo_icon_light_theme.png';
+import logo_text_dark_theme from '../../assets/images/logo_text_dark_theme.png';
+import logo_text_light_theme from '../../assets/images/logo_text_light_theme.png';
```
Nie używamy sneakcase, raczej w js - `camelCase`

TheFenix2000 commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+import '../MobileNavbar/MobileNavbar.scss';
+
+import Brightness6RoundedIcon from '@mui/icons-material/Brightness6Rounded';
+import { IconButton } from '@mui/material';
```
Użyć svg zamiast MUI

TheFenix2000 commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+      <div className="SidebarContainer">
+        <div className="Sidebar">
+          <IconButton>
+            <svg
```
svg importujemy do pliku i do folderu assets, zobacz inne PR

TheFenix2000 commented on src/components/MobileNavbar/MobileNavbar.scss:
```diff
@@ -0,0 +1,88 @@
+    position: -webkit-sticky;
+    position: sticky;
+    overflow: hidden;
+    font-family:  Helvetica;
```
font jest ustawiony dla całego body w baseStyle już

szymonbrud commented on src/components/MobileNavbar/MobileNavbar.js:
```diff
@@ -0,0 +1,93 @@
+      </div>
+      <div className="Language">
+        <div style={{ padding: '0.5rem' }}>
+          <a id="PL">PL</a>
```
Zmień to prosze na button, zarówno PL jak i EN

Commit:
fix: all comments resolved except mui icons

Commit:
fix: all comments resolved except mui icons

Commit:
fix: change IconButtons to buttons

Commit:
fix: change IconButtons to buttons

Commit:
fix: package jason + mobile navbar del

Commit:
fix: package jason + mobile navbar del

Commit:
fix: del style mobilenavbar

Commit:
fix: del style mobilenavbar

seba1288 committed:
Delete src/components/DesktopNavbar/Navbar1.scss

seba1288 committed:
Delete src/components/DesktopNavbar/Navbar1.scss

Commit:
fix: mui del

Commit:
fix: mui del

seba1288 committed:
Delete package.json

seba1288 committed:
Delete package-lock.json

seba1288 committed:
Add files via upload

TheFenix2000 commented on src/assets/styles/theme.scss:
```diff
@@ -9,6 +9,12 @@ $breakpoint-tablet: 700px;
 
+$dark-navbar-mobile:rgba(8,0,16);
+
+$grey_25: #999999;
```
jak nazwa tego koloru istnieje a istnieje niżej to układamy kolorki wg numeru od najmniejszego

TheWoxPL committed:
feature: add mobile sidebar menu 

- created SidebarMenu dictionary with SidebarMenu.js and SidebarMenu.module.scss
- added consts to utlis/constants/links.js
- added new color variable to theme.scss
- added 4 svg icons to assets/svgs

TheFenix2000 commented on src/components/Navbar/Navbar.js:
```diff
@@ -0,0 +1,82 @@
+import ThemeIconLight from '../../assets/svgs/ThemeIcon_dark.svg';
+import ThemeIconDark from '../../assets/svgs/ThemeIcon_light.svg';
+import {
+  ENG,
```
te klasy drukowane nei są poprawne, używamy camelcase czyli Eng i Pl

TheFenix2000 commented on src/components/Navbar/Navbar.module.scss:
```diff
@@ -0,0 +1,95 @@
+@import "../../assets/styles/theme";
```
przeformatuj, są dziwne odstępy

TheFenix2000 commented on src/components/Navbar/Navbar.js:
```diff
@@ -0,0 +1,82 @@
+import { StaticImage } from 'gatsby-plugin-image';
```
przeformatuj dla pewności przez ctrl+shift+p format document (forced) zobacz czy masz domyślny formater na prettiera

seba1288 replied:
powinienem miec bo zmienialismy to z kamilem
I czy nadal mam te zmiany wprowadzić jak juz z mergowaliscie?

Commit:
fix: add DS_Store to ignore

Commit:
Merge branch 'feature/mobile-navbar' of https://github.com/UEK-i-team/team-homepage into feature/mobile-navbar

Commit:
Revert "feature: add mobile sidebar menu "

This reverts commit 88436e4b94948c44a08b2c6238a927d1ad1091f0.

Commit:
fix: delted unused file

Commit:
fix: move grey color , format both documents ,chage camel case

kamil-ciesla commented on src/components/Navbar/Navbar.js:
```diff
@@ -0,0 +1,82 @@
+import ThemeIconLight from '../../assets/svgs/ThemeIcon_dark.svg';
+import ThemeIconDark from '../../assets/svgs/ThemeIcon_light.svg';
+import {
+  engLight,
+  LanguageThemeWraper,
+  Line,
+  LineDark,
+  Logo,
+  LogoContainer,
+  MainContainer,
+  MainContainerDark,
+  Name,
+  plDark,
+  plLight,
+  Sidebar,
+  SidebarContainer,
```
change to camelCase

kamil-ciesla commented on src/components/Navbar/Navbar.js:
```diff
@@ -0,0 +1,82 @@
+} from './Navbar.module.scss';
+
+export function Navbar() {
+  const [isLightTheme, setisLightTheme] = useState(true);
```
fix setIsLightTheme

Commit:
fix: chage to camel case

Commit:
fix: fix setIsLightTheme

Commit:
fix: more camel case
